hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/AccountFeature.java

```diff
@@ -148,4 +181,92 @@ public void verifyMirrorAPICryptoTransferResponse(int status) {
 
         assertThat(transferSum).isZero();
     }
+
+    @Then("the mirror node REST API should confirm the approved {long} tℏ crypto transfer allowance")
+    public void verifyMirrorAPIApprovedCryptoTransferResponse(long approvedAmount) {
+        verifyMirrorTransactionsResponse(mirrorClient, HttpStatus.OK.value());
+
+        String ownerString = ownerAccountId.toString();
+        String spenderString = senderAccountId.getAccountId().toString();
+        MirrorCryptoAllowanceResponse mirrorCryptoAllowanceResponse =
+                mirrorClient.getAccountCryptoAllowanceBySpender(ownerString, spenderString);
+
+        // verify valid set of allowance
+        List<MirrorCryptoAllowance> allowances = mirrorCryptoAllowanceResponse.getAllowances();
+        assertNotNull(allowances);
+        assertThat(allowances).isNotEmpty();
+
+        // verify transaction details
+        MirrorCryptoAllowance cryptoAllowance = allowances.get(0);
+
+        // verify valid allowance
+        assertNotNull(cryptoAllowance);
+        assertThat(cryptoAllowance.getAmount()).isEqualTo(approvedAmount);
+        assertThat(cryptoAllowance.getOwner()).isEqualTo(ownerString);
+        assertThat(cryptoAllowance.getPayerAccountId()).isEqualTo(ownerString);
+        assertThat(cryptoAllowance.getSpender()).isEqualTo(spenderString);
+        var timestampRange = cryptoAllowance.getTimestamp();
+        assertNotNull(timestampRange);
+        assertThat(timestampRange.getFrom()).isNotBlank();
+        assertThat(timestampRange.getTo()).isBlank();
```

Comment:
Use chaining to reduce overall lines of code and to return multiple errors at the same time.

```suggestion
        assertThat(mirrorCryptoAllowanceResponse.getAllowances())
            .isNotEmpty()
            .first()
            .isNotNull()
            .returns(approvedAmount, MirrorCryptoAllowance::getAmount)
            ...
            .extracting(MirrorCryptoAllowance::getTimestamp)
            .isNotNull()
            .satisfies(t -> assertThat(t.getFrom()).isNotBlank())
            .satisfies(t -> assertThat(t.getTo()).isBlank());
```

Reply:
Chained